test/decode_cms.py

```diff
@@ -0,0 +1,217 @@
+#!/usr/bin/env python3
+
+import inspect
+from datetime import datetime
+
+from pyasn1.codec.der import decoder
+from pyasn1.type import univ, char
+from pyasn1_modules import rfc2315, rfc5280, rfc5480, rfc5652, rfc5751
+
+
+def build_oid_map(modules):
+    oid_map = {}
+    for module in modules:
+        for name, obj in inspect.getmembers(module):
+            if isinstance(obj, univ.ObjectIdentifier):
+                oid_str = str(obj)
+                name = name.removeprefix("id_")
+                oid_map[oid_str] = name
+    return oid_map
+
+
+OID_MAP = build_oid_map([rfc2315, rfc5280, rfc5480, rfc5652, rfc5751])
+
+
+def resolve_oid(oid):
+    return OID_MAP.get(str(oid), oid)
+
+
+def decode_as_string(value_any):
+    for asn1_type in [char.PrintableString, char.UTF8String]:
+        try:
+            decoded, _ = decoder.decode(value_any.asOctets(), asn1Spec=asn1_type())
+            return str(decoded)
+        except Exception:
+            continue
+    # fallback: return hex
+    return value_any.prettyPrint()
+
+
+def name_to_string(name):
+    parts = []
+    seq = name.getComponentByName("rdnSequence")
+    for rdn in seq:  # each rdn is a SET OF AttributeTypeAndValue
+        if isinstance(rdn, str):
+            continue
+        for attr in rdn:
+            type_oid = attr.getComponentByName("type")
+            value = attr.getComponentByName("value")
+            value_str = decode_as_string(value)
+
+            oid_map = {
+                "2.5.4.3": "CN",
+                "2.5.4.10": "O",
+            }
+
+            oid_str = str(type_oid)
+            name = oid_map.get(oid_str, oid_str)
+
+            parts.append(f"{name}={value_str}")
+
+    return ", ".join(parts)
+
+
+def utctime_to_datetime(utctime):
```

Comment:
Maybe add a note that this decodes `YYMMDDHHMMSSZ` and the specialty of `YY` to ease following the code, e.g.

```suggestion
def utctime_to_datetime(utctime):
    """
    Decode an ASN.1 UTCTime string in YYMMDDHHMMSSZ format to a datetime object.
    Assumes:
      - YY in [00,49] => 2000..2049
      - YY in [50,99] => 1950..1999 (ASN.1 UTCTime rule)
    """
```